### ISSUE TITLE
Add auto-update support

### ISSUE THREAD

**Issue by mwbrooks** [phonegap/phonegap-app-desktop#232]
We should consider adding auto-update support. When a newer version of the application is available, we can download it, install it, and reopen the application automatically. This is pretty standard in most modern desktop apps.

By the looks of it, node-webkit defers this responsibility to: https://github.com/edjafarov/node-webkit-updater

It looks like it's an active project that's well designed and easy to use. I'm not sure how well it works though.

**Comment by hollyschinsky** [comment]
Getting some issues with auto-update when going from 0.1.3 to the latest version (0.1.4). Tried on two machines, with the following results: 

When I run this on a laptop running Yosemite (10.10.2), the green Update button downloads the app to a temp folder (to Macintosh HD ▸ private ▸ var ▸ folders ▸ _x ▸ s6wsj5sd659d0fp45dbnyd_m0000gn ▸ T ▸ PhoneGap-Desktop-Beta-0.1.4-mac) but doesn't complete the installation when I click Restart. It appears to try to open and run it but stays blank then closes on it's own. 

When I tried on another laptop running OS X 10.9.2 it downloads the file (to Macintosh HD ▸ private ▸ var ▸ folders ▸ _x ▸ s6wsj5sd659d0fp45dbnyd_m0000gn ▸ T ▸ PhoneGap-Desktop-Beta-0.1.4-mac) and opens it when I click Restart, but it doesn't move it to the Applications folder or overwrite the old version so I would have to go find it again to open it otherwise I still get the old one.

**Comment by hermwong** [comment]
It looks like the update is being properly downloaded and unzipped in the temp folder, but when the restart happens the PG-Desktop opens with a blank screen and the updated version doesn't seem to get copied over to the Applications folder (tested on Yosemite OSX 10.10.2).

When the restart is triggered from the Restart button it tries to run the updated version of PG-Desktop from the temp folder. If I navigate to the temp folder in Finder and click on the downloaded update the PG-Desktop does load properly. I still need to determine reason for the blank screen when the restart is triggered from the PG-Desktop.

Basically there are 2 issues that are causing the auto-update to fail:
 1. Restart doesn't work when triggered from PG-Desktop
 1. Updated PG-Desktop isn't being copied over to the Applications folder.

**Comment by hermwong** [comment]
Discovered that the update will fail if the user goes through the auto-update workflow which fails the restart and then attempts to auto-update again when they manually restart.

It seems that there is an issue / conflict when attempting to re-download the update when the user attempts the second auto-update because the `zip` already exists in the temp folder.
